### ISSUE TITLE
Magazine reliability

### ISSUE THREAD

**Issue by kevingranade** [CleverRaven/Cataclysm-DDA#30395]
**Is your feature request related to a problem? Please describe.**  
All magazines, on-brand, off brand, high tech, makeshift etc all have the same perfect degree of reliability.

**Describe the solution you'd like**  
Failure to feed feature for magazines, leading to jams, scaled based on configured reliability of each magazine.
Like with guns, magazines are intended to have a reliability score indicating how likely they are to malfunction, but this was either removed or never completed.
Each time a gun action cycles (after firing instead of before, ideally), there is a chance based on magazine properties and possibly condition that the gun will fail to cycle properly.  This can be lumped together into a "jam" for the initial version, which prevents firing the gun, and requires the player to take action to clear the jam.  Once the jam is cleared, the round would generally land on the ground under the player.

**Deferred work**
There are a number of conditions that could be tracked to add additional nuance to magazine reliability.
Overloading magazines increases likelihood of a jam.
Damage to magazines can drastically increase jam chance.
Dirt in the magazine can drastically increase chances of jamming.
Certain combinations of gun and magazine might be particularly bad.
Individual magazines from an otherwise bad production run might have acceptable reliability.
Some jams or other malfunctions can be more severe, leading to persistent disabling of the gun until it can be worked on.

**Additional context**  
This came up while reviewing #30326, which is more critical than existing magazine handling because a key property of these new magazines is their unreliability.
#14993 was closed because no one seemed interested in implementing this mechanism, partially because of it's rarity with professionally constructed magazines. With the advent of player-constructed magazines, the possibility of extremely frequent and inpactful jams becomes a real possibility.
